reposilite-frontend/src/store/log.js

```diff
@@ -1,78 +1,58 @@
-import { ref, watch, watchEffect, reactive } from 'vue'
+import { computed, ref, watch, watchEffect, reactive } from 'vue'
 import Convert from 'ansi-to-html'
 
 const levelNames = ['Trace', 'Debug', 'Info', 'Warn', 'Error']
 const levels  = reactive({})
 const filter = ref('')
 const id = ref(0)
-const rawLog = ref([])
-const log = ref([])
+const rawLog = reactive([])
 const convert = new Convert()
 
+const getLevel = message => {
+  for (const level of levelNames) {
+    // NOTE: If the message starts with following prefix, then use startsWith
+    if (message.includes(`&nbsp;|&nbsp;${level.toUpperCase()}&nbsp;|&nbsp;`)) {
+      return level
+    }
+  }
+}
+
+const sanitizeMessage = (message) => convert.toHtml(
+  message
+    .replaceAll('<', '&lt;')
+    .replaceAll('>', '&gt;')
+    .replaceAll(' ', '&nbsp;')
+)
+
 export default function useLog() {
-  levelNames.forEach(level => {
+  for (const level of levelNames) {
     levels[level] = {
       name: level,
       enabled: true,
-      count: 0,
-      test: (message) => message.includes(`&nbsp;|&nbsp;${level.toUpperCase()}&nbsp;|&nbsp;`)
+      count: computed(() => rawLog.reduce((accumulator, entry) => accumulator + (entry.level === level), 0))
```

Comment:
You can optimize this by making a global computed like:
```js
const count = computed(() => rawLog.reduce((accumulator, entry) => {
  accumulator[entry.level] ?= 0
  accumulator[entry.level] += 1
  return accumulator
}, {}))
```

And using: 
```
count: count[level]
```

Reply:
The `variable ?= value` syntax is setting the value of `variable` to  value of `value` only if the current `variable` value is nullish. In case you didn't know.